registry/server/api_topics.go

```diff
@@ -185,25 +185,37 @@ func (s *Server) CreateTopic(ctx context.Context, req *pb.CreateTopicRequest) (*
 	// If we're targeting a specific set of brokers by tag, build
 	// a replica assignment.
 	var assignment kafkaadmin.ReplicaAssignment
-	if req.TargetBrokerTags != nil {
+	if req.TargetBrokerTags != nil || req.TargetBrokerIds != nil {
 		// Create a stub map with the provided request dimensions.
 		opts := kafkazk.Populate(
 			req.Topic.Name,
 			int(req.Topic.Partitions),
 			int(req.Topic.Replication),
 		)
 		pMap := kafkazk.NewPartitionMap(opts)
-
 		// Fetch brokers by tag.
 		reqParams := &pb.BrokerRequest{Tag: req.TargetBrokerTags}
 		resp, err := s.ListBrokers(ctx, reqParams)
 		if err != nil {
 			return empty, err
 		}
 
-		targetBrokerIDs := make([]int, len(resp.Ids))
-		for i := range resp.Ids {
-			targetBrokerIDs[i] = int(resp.Ids[i])
+		selectedBrokerIds := make(map[uint32]bool)
```

Comment:
[nit] I think in this code base, the more idiomatic way to emulate a set would be `map[uint32]struct{}`.

Reply:
Background on that: `struct{}` types are the smallest memory footprint + set membership only relies on the map key presence (the map semantics  can be confusing here, when you do an assignment that's _not_ in a `range`, many people assume that it's `k, v := m[key]`. What it's actually returning is `{key, <bool as to whether the key is in the map>`}. So you can use any value you want for sets: `m[key] = "banana"; if _, exists := m[key]; exists {do stuff}`).